crates/pallet-native-to-evm-currency-swap/src/lib.rs

```diff
@@ -0,0 +1,185 @@
+    }
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// Balances were swapped.
+        BalancesSwapped {
+            /// The account id balances withdrawed from.
+            from: T::AccountId,
+            /// The withdrawed balances amount.
+            withdrawed_amount: NativeBalanceOf<T>,
+            /// The account id balances deposited to.
+            to: T::EvmAccountId,
+            /// The deposited balances amount.
+            deposited_amount: EvmBalanceOf<T>,
+            /// The corresponding transaction hash executed in evm.
+            evm_trx_hash: H256,
+        },
+    }
+
+    #[pallet::call(weight(<T as Config>::WeightInfo))]
+    impl<T: Config> Pallet<T> {
+        /// Swap balances.
+        #[pallet::call_index(0)]
+        pub fn swap(
+            origin: OriginFor<T>,
+            to: T::EvmAccountId,
+            #[pallet::compact] amount: NativeBalanceOf<T>,
+        ) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+
+            with_storage_layer(move || {
+                Self::do_swap(who, to, amount, Preservation::Expendable)?;
+
+                Ok(())
+            })
+        }
+
+        /// Same as the swap call, but with a check that the swap will not kill the origin account.
+        #[pallet::call_index(1)]
+        pub fn swap_keep_alive(
+            origin: OriginFor<T>,
+            to: T::EvmAccountId,
+            #[pallet::compact] amount: NativeBalanceOf<T>,
+        ) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+
+            with_storage_layer(move || {
+                Self::do_swap(who, to, amount, Preservation::Preserve)?;
+
+                Ok(())
+            })
+        }
+    }
+
+    impl<T: Config> Pallet<T> {
+        /// General swap balances implementation.
+        pub fn do_swap(
+            who: T::AccountId,
+            to: T::EvmAccountId,
+            amount: NativeBalanceOf<T>,
+            preservation: Preservation,
+        ) -> DispatchResult {
+            let estimated_swapped_balance = T::BalanceConverter::convert(amount);
+            T::EvmCurrency::can_deposit(&to, estimated_swapped_balance, Provenance::Extant)
+                .into_result()?;
+
+            T::NativeCurrency::transfer(&who, &T::PotNativeBridge::get(), amount, preservation)?;
+
+            let evm_balance_to_be_deposited: u128 =
+                estimated_swapped_balance.unique_saturated_into();
+
+            let transaction = pallet_ethereum::Transaction::EIP1559(ethereum::EIP1559Transaction {
+                chain_id: <T as pallet_evm::Config>::ChainId::get(),
+                nonce: pallet_evm::Pallet::<T>::account_basic(&T::PotEvmBridge::get().into())
+                    .0
+                    .nonce,
+                max_priority_fee_per_gas: 0.into(),
+                max_fee_per_gas: 0.into(),
+                gas_limit: 21000.into(), // simple transfer
```

Comment:
This should be taken from the `EvmConfig` or whatever that is called; the rationale is this value can change across EVM versions (i.e. London vs Istambul, that sort of thing).
Either expose this as a config parameter of the pallet itself, or just directly instantiate from the EVM config.

Reply:
Extracted `gas_limit` value from `EvmConfig` - https://github.com/humanode-network/humanode/pull/1449/commits/f48f89ea108e58aff831a9f014459b0795e0ac41